numpy/core/_methods.py

```diff
@@ -242,10 +264,27 @@ def _std(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
 
     return ret
 
-def _ptp(a, axis=None, out=None, keepdims=False):
+def _ptp(a, axis=None, out=None, keepdims=False, *, where=True):
+    max_call = (a, axis, None, out, keepdims)
+    min_call = (a, axis, None, None, keepdims)
+
+    if where is not True:
+        # guarded to protect circular imports
+        from numpy.core.getlimits import iinfo, finfo
+
+        if nt._kind_name(a.dtype) in ['uint', 'int']:
+            _info = iinfo(a.dtype)
+        elif nt._kind_name(a.dtype) in ['float', 'complex']:
+            _info = finfo(a.dtype)
+        else:
+            raise TypeError(('Cannot evaluate peak-to-peak for dtypes other '
+                             'than uint, int, float or complex'))
+        max_call += (_info.min, where)
+        min_call += (_info.max, where)
+
     return um.subtract(
-        umr_maximum(a, axis, None, out, keepdims),
-        umr_minimum(a, axis, None, None, keepdims),
+        umr_maximum(*max_call),
+        umr_minimum(*min_call),
         out
     )
```

Comment:
This doesn't do the right thing for arrays containing `np.inf`. I recommend we skip `ptp` for now, and attempt it in a future PR - it's sufficiently different from `mean` and `var` that I think it's worth handling separately, so that we can get the rest merged.

Reply:
Alright, I will remove changes made to `ptp` and upload a commit without it.